### ISSUE TITLE
file.prepend state fails on non-existent file when makedirs is False

### ISSUE THREAD

**Issue by Deewiant** [saltstack/salt#27401]
This is basically #1864 but for `file.prepend` instead of `file.append`: if the file to be prepended to does not exist, Salt throws an error.

The difference is quite visible in the code of `salt/states/file.py`. The definition of `append` has a block like this:

```python
    check_res, check_msg = _check_file(name)
    if not check_res:
        touch(name, makedirs=makedirs)
        retry_res, retry_msg = _check_file(name)
        if not retry_res:
            return _error(ret, check_msg)
```

Whereas in `prepend` it looks like this:

```python
    check_res, check_msg = _check_file(name)
    if not check_res:
        return _error(ret, check_msg)
```

I suspect the `prepend` one should look identical to the `append` one. (There's also a misleading call `__salt__['file.touch'](name)` immediately above this code in each function, but that's only used when `makedirs` has been set.)

**Comment by jfindlay** [comment]
@Deewiant, thanks for the report.  You are welcome to submit a pull request against the 2015.5 branch.

**Comment by rallytime** [comment]
@Deewiant Since you fix has been merged (#27441) do you consider this issue resolved?

**Comment by Deewiant** [comment]
Yes, indeed. Closing.
